conda/cli/install.py

```diff
@@ -172,119 +170,201 @@ def get_revision(arg, json=False):
-        else:
-            validate_prefix_is_writable(prefix)
     else:
+        # if this is not a new env and the prefix does not exist, then
+        # there is no existing environment - this is an error
         raise EnvironmentLocationNotFound(prefix)
 
-    args_packages = [s.strip("\"'") for s in args.packages]
-    if newenv and not args.no_default_packages:
-        # Override defaults if they are specified at the command line
-        names = [MatchSpec(pkg).name for pkg in args_packages]
-        for default_package in context.create_default_packages:
-            if MatchSpec(default_package).name not in names:
-                args_packages.append(default_package)
 
-    context_channels = context.channels
-    index_args = {
+def get_index_args(args) -> dict[str, any]:
+    """Returns a dict of args required for fetching an index
+
+    :param args: The args provided by the cli
+    :returns: dict of index args
+    """
+    return {
         # TODO: deprecate --use-index-cache
         # "use_cache": args.use_index_cache,  # --use-index-cache
-        "channel_urls": context_channels,
+        "channel_urls": context.channels,
         # TODO: deprecate --unknown
         # "unknown": args.unknown,  # --unknown
         "prepend": not args.override_channels,  # --override-channels
         "use_local": args.use_local,  # --use-local
     }
 
+
+def ensure_update_specs_exist(prefix: str, specs: list[str]):
+    """Checks that each spec that is requested as an update exists in the prefix
+
+    :param prefix: The target install prefix
+    :param specs: List of specs to be updated
+    :raises CondaError: if there is an invalid spec provided
+    :raises PackageNotInstalledError: if the requested specs to install don't exist in the prefix
+    """
+    prefix_data = PrefixData(prefix)
+    for spec in specs:
+        spec = MatchSpec(spec)
+        if not spec.is_name_only_spec:
+            raise CondaError(
+                f"Invalid spec for 'conda update': {spec}\nUse 'conda install' instead."
+            )
+        if not prefix_data.get(spec.name, None):
+            raise PackageNotInstalledError(prefix, spec.name)
+
+
+def install(args, parser, command="install"):
+    """Logic for `conda install`, `conda update`, and `conda create`."""
+    prefix = context.target_prefix
+    newenv = bool(command == "create")
+    isupdate = bool(command == "update")
+    isinstall = bool(command == "install")
+    isremove = bool(command == "remove")
+
+    # common validations for all types of installs
+    validate_install_command(prefix=prefix, newenv=newenv)
+
+    # this is sort of a hack.  current_repodata.json may not have any .tar.bz2 files,
+    #    because it deduplicates records that exist as both formats.  Forcing this to
+    #    repodata.json ensures that .tar.bz2 files are available
+    if context.use_only_tar_bz2:
+        args.repodata_fns = ("repodata.json",)
```

Comment:
We should only override if the user didn't provide their own. e.g. it would be surprising if we do this when they provided `--repodata-fn=my-custom-repodata.json`.

Reply:
I think this would be a pretty big change in behaviour and not appropriate to be in a refactor PR. For reference, compare this to the main branch https://github.com/conda/conda/blob/main/conda/cli/install.py#L182

Can you create a seperate issue (or is this a bug?) to address this.

Reply:
I'll raise an issue and assign you, thanks.

Reply:
https://github.com/conda/conda/issues/14745 - can't assign you until you comment there, though.